olive/engine/engine.py

```diff
@@ -23,13 +23,15 @@
 from olive.exception import EXCEPTIONS_TO_RAISE, OlivePassError
 from olive.hardware import AcceleratorLookup, AcceleratorSpec, Device
 from olive.model import ModelConfig
-from olive.passes.olive_pass import Pass
 from olive.strategy.search_strategy import SearchStrategy
 from olive.systems.common import SystemType
 from olive.systems.local import LocalSystem
 from olive.systems.olive_system import OliveSystem
 from olive.systems.utils import create_new_system_with_cache
 
+if TYPE_CHECKING:
```

Comment:
nit: are these changes also needed?

Reply:
Or this method is recommended since we are only using `Pass` for type hints?

Reply:
I add the TYPE_CHECKING here to make sure the explicit Pass import is optional for engine running.

Reply:
Sounds good to me. Thanks!

Reply:
as for best practice, any hint import should be put into TYPE_CHECKING.